### PR TITLE
[CSE-74] Refactor Search UI

### PR DESCRIPTION
**Pull request by sectore**
... to have it on any pages (desktop version) and at the overlay menu (mobile version)

This PR includes also some style fixes.

### CODE REVIEW

File: frontend/src/Explorer/View/Layout.purs [hunk 1]
```diff
@@ -24,17 +25,18 @@ view state =
     let mobileMenuClazz = if state ^. (viewStates <<< globalViewState <<< gViewMobileMenuOpenend)
                           then " mobile__menu--opened"
                           else ""
+        routeClazz = CSS.route $ state ^. route
     in
     P.div
-      [ P.className $ "explorer-container" <> mobileMenuClazz ]
+      [ P.className $ "explorer-container" <> mobileMenuClazz <> " " <> routeClazz ]
```
Comment (akegalj) [inline frontend/src/Explorer/View/Layout.purs]: This looks very fragile. I would much rather wanted to see:
```
P.className $ String.joinWith " " ["explorer-container", mobileMenuClazz, routeClazz]
```

It makes clear this way we won't accidentally concatenate two classes.

File: frontend/src/Explorer/View/Search.purs [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+inputSlotName = "inp_slot"
+
+searchInputView :: State -> P.Html Action
+searchInputView state =
+    let lang' = state ^. lang
+        dbViewSearchInputFocused = state ^. (viewStates <<< globalViewState <<< gViewSearchInputFocused)
+        searchIconClazz = if dbViewSearchInputFocused then " bg-icon-search-hover" else " bg-icon-search"
+        selectedSearch = state ^. (viewStates <<< globalViewState <<< gViewSelectedSearch)
+        addrHiddenClazz = if selectedSearch == SearchTime  then " hide " else ""
+        epochHiddenClazz = if selectedSearch /= SearchTime  then " hide " else ""
```
Comment (akegalj) [inline frontend/src/Explorer/View/Search.purs]: Similar as previous commit. I would like classes not to be manually wrapped with extra space but would prefer using `"someClass"` and then `String.joinWith " "` for concatenating those. Also adding more type safety (like newtype wrapper) around classes would be nice.

File: frontend/src/Explorer/View/Search.purs [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+                              else translate (I18nL.hero <<< I18nL.hrSearch) lang'
+            , P.onFocus <<< const $ GlobalFocusSearchInput true
+            , P.onBlur <<< const $ GlobalFocusSearchInput false
+            , P.onChange $ GlobalUpdateSearchValue <<< _.value <<< _.target
+            , P.onKey "enter" $ const GlobalSearch
+            , P.value $ state ^. (viewStates <<< globalViewState <<< gViewSearchQuery)
+            ]
+            []
+        , P.div
+            [ P.className $ "explorer-search__wrapper" <> epochHiddenClazz ]
```
Comment (akegalj) [inline frontend/src/Explorer/View/Search.purs]: For example, from this line it is not clear did we accidentally concatenate these two classes. With `String.joinWith " " [...]` I would be sure and won't have to look to classes to see do they have extra space or not.

**Comment by sectore** [comment]
@akegalj Thanks for your review. I agree in all CSS related concerns you have had. With `CSE-78` we will have an issue for this.

**Comment by sectore** [comment]
> You can see these small issues when you lower the width below 350 px.

@ksaric Thanks for your review. I don't know if we really support this small devices. It might be a question to our design team.

**Pull request merged**